src/main/java/com/socrata/api/Soda2Producer.java

```diff
@@ -76,6 +78,7 @@ public static final Soda2Producer newProducerWithRequestId(final String url, Str
     public Soda2Producer(HttpLowLevel httpLowLevel)
     {
         super(httpLowLevel);
+        factory = new JsonFactory();
```

Comment:
@gilday Alright, your user has rights to create datasets on https://sandbox.demo.socrata.com, let me know how the integration testing goes.